### ISSUE TITLE
Add support for formatting scaladoc (markdown) code blocks 

### ISSUE THREAD

**Issue by tdudzik** [scalameta/scalafmt#2498]
We would like to add support for formatting code blocks in scaladoc. We would like to format especially markdown code blocks (i.e. code blocks between ` ```scala ` and ` ``` `), however, I think it should be considered if code blocks between `{{{` and `}}}` should be formatted as well.

E.g. scaladoc like this:
````scala
/**
  *  ```scala
  *    object Foo { def bar(): String = "foobar" }
  *  ```
  */
````

should be formatted to scaladoc like this:
````scala
/**
  *  ```scala
  *    object Foo {
  *      def bar(): String = "foobar" 
  *    }
  *  ```
  */
````

Also, considering new scaladoc features that are currently under development, there can be some options after the code block prefix, e.g.:  
````scala
/**
  *  ```scala sc:compile 
  *    object Foo {
  *      def bar(): String = "foobar" 
  *    }
  *  ```
  */
````

It would require some work in scalafmt and scalameta. Scalameta part is currently being developed by me here: https://github.com/scalameta/scalameta/pull/2332. This issue can be used to discuss potential solutions and to decide if this feature can/should be implemented at all.

**Comment by tgodzik** [comment]
@kitbellew before merging scalameta PR, we could experiment with blocks produced with `{{{` and see if it's actually plausible. Do you think taking a similar approach as with Ammonite scripts makes sense? We would treat the code blocks as separate files. This is just my guess.

**Comment by kitbellew** [comment]
fixed in #2503.
